core/src/test/java/nl/nn/adapterframework/util/FileUtilsTest.java

```diff
@@ -0,0 +1,283 @@
+package nl.nn.adapterframework.util;
+
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * FileUtils Tester.
+ *
+ * @author <Sina Sen>
+ */
+public class FileUtilsTest {
+
+	@ClassRule
+	public static TemporaryFolder testFolderSource = new TemporaryFolder();
+
+	@ClassRule
+	public static TemporaryFolder testFolderDest = new TemporaryFolder();
+
+	public static String sourceFolderPath;
+
+	public static String destFolderPath;
+
+	private static File f1;
+
+	public String sep = File.separator;
+
+	@BeforeClass
+	public static void setUpTest() throws IOException {
+		sourceFolderPath = testFolderSource.getRoot().getPath();
+		f1 = testFolderSource.newFile("1.txt");
+		destFolderPath = testFolderDest.getRoot().getPath();
+	}
+
+
+	@Test
+	public void testGetFreeFile() throws Exception {
+		File f = new File(".." + sep + "core" + sep + "src" + sep + "test" + sep + "resources" + sep + "Pipes");
+		File res = FileUtils.getFreeFile(f);
+		assertEquals("Pipes", res.getName());
+	}
+
+
+	@Test
+	public void testMoveFile() throws Exception {
+		File toBeMoved = new File(".." + sep + "core" + sep + "src" + sep + "test" + sep + "resources" + sep + "Pipes" + sep + "movingFile.txt");
```

Comment:
Please use an URL and turn that into a file here. Renaming the path or whatever would otherwise needlessly break the test.
`this.getClass().getResource("/Pipes/movingFile.txt");` will return an URL (to which you could apply a null check (`assertNotNull(xxxx)`) which you can then feed to the new File constructor :)

Reply:
I just realized you do this more often, maybe you could create a method that does all this and returns a `File`?

Reply:
yes, I will create a method for this and replace all.